runtime/src/iree/hal/drivers/cuda/cuda_allocator.c

```diff
@@ -155,15 +155,19 @@ iree_hal_cuda_allocator_query_compatibility(
 }
 
 static void iree_hal_cuda_buffer_free(iree_hal_cuda_context_wrapper_t* context,
-                                      iree_hal_memory_type_t memory_type,
+                                      iree_hal_cuda_buffer_type_t buffer_type,
                                       CUdeviceptr device_ptr, void* host_ptr) {
   IREE_TRACE_ZONE_BEGIN(z0);
-  if (iree_all_bits_set(memory_type, IREE_HAL_MEMORY_TYPE_DEVICE_LOCAL)) {
-    // Device local.
-    CUDA_IGNORE_ERROR(context->syms, cuMemFree(device_ptr));
-  } else {
-    // Host local.
-    CUDA_IGNORE_ERROR(context->syms, cuMemFreeHost(host_ptr));
+  switch (buffer_type) {
+    case IREE_HAL_CUDA_BUFFER_TYPE_DEVICE:
```

Comment:
I think you need braces around the case clauses because the error macros have a side-effect of introducing new variables and some compilers don't like that.